diff --git a/LexLPeg.cxx b/LexLPeg.cxx
index eb96f4a..5f67bac 100644
--- a/LexLPeg.cxx
+++ b/LexLPeg.cxx
@@ -802,7 +802,7 @@ void SCI_METHOD LexerLPeg::Fold(
 }
 
 Sci_Position SCI_METHOD LexerLPeg::PropertySet(const char *key, const char *value) {
-  props.Set(key, value);
+  props.Set(key, value, strlen(key), strlen(value));
   if (strcmp(key, LexerHomeKey) == 0 && lexerNames.empty())
     ReadLexerNames(value); // not using SCI_CREATELOADER private call
   if (reinit && (strcmp(key, LexerHomeKey) == 0 || strcmp(key, LexerNameKey) == 0)) Init();
@@ -939,11 +939,16 @@ LexerModule lmLPeg(SCLEX_LPEG, LexerLPeg::LexerFactoryLPeg, "lpeg");
 #define EXPORT_FUNCTION __declspec(dllexport)
 #define CALLING_CONVENTION __stdcall
 #else
+#if (_MSC_VER)
+#define EXPORT_FUNCTION
+#define CALLING_CONVENTION
+#else
 #define EXPORT_FUNCTION __attribute__((visibility("default")))
 #define CALLING_CONVENTION
+#endif
 #endif // _WIN32
 
-extern "C" {
+namespace Scintillua {
 
 /** Returns 1, the number of lexers defined in this file. */
 EXPORT_FUNCTION int CALLING_CONVENTION GetLexerCount() { return 1; }
@@ -1018,7 +1023,7 @@ EXPORT_FUNCTION ILexer5 *CALLING_CONVENTION CreateLexer(const char *name) {
   }
   return lpegLexer;
 }
-}
+} // namespace Scintillua
 
 /*
 Forward the following properties from SciTE.
diff --git a/LexLPeg.h b/LexLPeg.h
index 4aed0c9..5eb439f 100644
--- a/LexLPeg.h
+++ b/LexLPeg.h
@@ -10,18 +10,19 @@
 #ifdef __cplusplus
 #include "ILexer.h"
 #define ILEXER5 Scintilla::ILexer5
-extern "C" {
+
 #else
 #define ILEXER5 void
 #endif
 
+namespace Scintillua {
 const char *GetLibraryPropertyNames();
 void SetLibraryProperty(const char *key, const char *value);
 const char *GetNameSpace();
 ILEXER5 *CreateLexer(const char *name);
-
+} // namespace Scintillua
 #ifdef __cplusplus
-}
+
 #endif
 
 #endif
