diff --git a/Scintillua.cxx b/Scintillua.cxx
index 48a3fc9..7f1bed0 100644
--- a/Scintillua.cxx
+++ b/Scintillua.cxx
@@ -496,15 +496,22 @@ const char *Scintillua::PropertyGet(const char *key) { return props.Get(key); }
 
 const char *Scintillua::GetName() { return name.c_str(); }
 
-#if (_WIN32 && !NO_DLL)
-#define EXPORT_FUNCTION __declspec(dllexport)
-#define CALLING_CONVENTION __stdcall
+} // namespace
+
+#if (WIN32)
+  #if (NO_DLL)
+    #define EXPORT_FUNCTION
+    #define CALLING_CONVENTION
+  #else
+    #define EXPORT_FUNCTION __declspec(dllexport)
+    #define CALLING_CONVENTION __stdcall
+  #endif
 #else
 #define EXPORT_FUNCTION __attribute__((visibility("default")))
 #define CALLING_CONVENTION
 #endif // _WIN32
 
-extern "C" {
+namespace ScintilluaNS {
 
 EXPORT_FUNCTION int CALLING_CONVENTION GetLexerCount() { return 1; }
 
@@ -540,6 +547,4 @@ EXPORT_FUNCTION Scintilla::ILexer5 *CALLING_CONVENTION CreateLexer(const char *n
 EXPORT_FUNCTION const char *CALLING_CONVENTION GetCreateLexerError() {
   return errorMessage.c_str();
 }
-}
-
-} // namespace
+}
\ No newline at end of file
diff --git a/Scintillua.h b/Scintillua.h
index fa006bc..7bcd060 100644
--- a/Scintillua.h
+++ b/Scintillua.h
@@ -7,19 +7,20 @@
 #ifdef __cplusplus
 #include "ILexer.h"
 #define ILEXER5 Scintilla::ILexer5
-extern "C" {
+
 #else
 #define ILEXER5 void
 #endif
 
+namespace ScintilluaNS {
 const char *GetLibraryPropertyNames();
 void SetLibraryProperty(const char *key, const char *value);
 const char *GetNameSpace();
 ILEXER5 *CreateLexer(const char *name);
 const char *GetCreateLexerError();
-
-#ifdef __cplusplus
 }
+#ifdef __cplusplus
+
 #endif
 
 #endif
