diff --git a/qt/ScintillaEdit/ScintillaEdit.pro b/qt/ScintillaEdit/ScintillaEdit.pro
index 4e2a9de7..4b55f2df 100644
--- a/qt/ScintillaEdit/ScintillaEdit.pro
+++ b/qt/ScintillaEdit/ScintillaEdit.pro
@@ -6,6 +6,7 @@
 
 QT       += core gui
 greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
+equals(QT_MAJOR_VERSION, 6): QT += core5compat
 
 TARGET = ScintillaEdit
 TEMPLATE = lib
diff --git a/qt/ScintillaEditBase/PlatQt.cpp b/qt/ScintillaEditBase/PlatQt.cpp
index 2d49dcb5..a72bba69 100644
--- a/qt/ScintillaEditBase/PlatQt.cpp
+++ b/qt/ScintillaEditBase/PlatQt.cpp
@@ -36,7 +36,9 @@
 #include <QListWidget>
 #include <QVarLengthArray>
 #include <QScrollBar>
+#if QT_VERSION < QT_VERSION_CHECK(5, 10, 0)
 #include <QDesktopWidget>
+#endif
 #include <QTextLayout>
 #include <QTextLine>
 #include <QLibrary>
@@ -627,7 +629,7 @@ XYPOSITION SurfaceImpl::WidthText(const Font *font, std::string_view text)
 	QFontMetricsF metrics(*FontPointer(font), device);
 	SetCodec(font);
 	QString su = UnicodeFromText(codec, text);
-	return metrics.width(su);
+	return metrics.horizontalAdvance(su);
 }
 
 void SurfaceImpl::DrawTextNoClipUTF8(PRectangle rc,
@@ -708,7 +710,7 @@ XYPOSITION SurfaceImpl::WidthTextUTF8(const Font *font, std::string_view text)
 {
 	QFontMetricsF metrics(*FontPointer(font), device);
 	QString su = QString::fromUtf8(text.data(), static_cast<int>(text.length()));
-	return metrics.width(su);
+	return metrics.horizontalAdvance(su);
 }
 
 XYPOSITION SurfaceImpl::Ascent(const Font *font)
@@ -930,7 +932,11 @@ public:
 protected:
 	void selectionChanged(const QItemSelection &selected, const QItemSelection &deselected) override;
 	void mouseDoubleClickEvent(QMouseEvent *event) override;
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+	void initViewItemOption(QStyleOptionViewItem *option) const override;
+#else
 	QStyleOptionViewItem viewOptions() const override;
+#endif
 
 private:
 	IListBoxDelegate *delegate;
@@ -1254,12 +1260,20 @@ void ListWidget::mouseDoubleClickEvent(QMouseEvent * /* event */)
 	}
 }
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+void ListWidget::initViewItemOption(QStyleOptionViewItem *option) const
+{
+	QListWidget::initViewItemOption(option);
+	option->state |= QStyle::State_Active;
+}
+#else
 QStyleOptionViewItem ListWidget::viewOptions() const
 {
 	QStyleOptionViewItem result = QListWidget::viewOptions();
 	result.state |= QStyle::State_Active;
 	return result;
 }
+#endif
 //----------------------------------------------------------------------
 Menu::Menu() noexcept : mid(nullptr) {}
 void Menu::CreatePopUp()
diff --git a/qt/ScintillaEditBase/PlatQt.h b/qt/ScintillaEditBase/PlatQt.h
index 02f3b0f7..a7c40515 100644
--- a/qt/ScintillaEditBase/PlatQt.h
+++ b/qt/ScintillaEditBase/PlatQt.h
@@ -28,6 +28,7 @@
 #include <QPaintDevice>
 #include <QPainter>
 #include <QHash>
+#include <QTextCodec>
 
 namespace Scintilla::Internal {
 
