diff --git a/Scintillua.cxx b/Scintillua.cxx
index e0bbf8f..5e3cd2c 100644
--- a/Scintillua.cxx
+++ b/Scintillua.cxx
@@ -523,20 +523,22 @@ const char *Scintillua::PropertyGet(const char *key) { return props.Get(key); }
 
 const char *Scintillua::GetName() { return name.c_str(); }
 
-#if _WIN32
-#if !NO_DLL
+} // namespace
+
+#if (WIN32)
+#if (NO_DLL)
+#define EXPORT_FUNCTION
+#define CALLING_CONVENTION
+#else
 #define EXPORT_FUNCTION __declspec(dllexport)
 #define CALLING_CONVENTION __stdcall
-#else
-#define EXPORT_FUNCTION extern
-#define CALLING_CONVENTION
 #endif
 #else
 #define EXPORT_FUNCTION __attribute__((visibility("default")))
 #define CALLING_CONVENTION
 #endif // _WIN32
 
-extern "C" {
+namespace ScintilluaNS {
 
 EXPORT_FUNCTION int CALLING_CONVENTION GetLexerCount() { return 1; }
 
@@ -572,6 +574,4 @@ EXPORT_FUNCTION Scintilla::ILexer5 *CALLING_CONVENTION CreateLexer(const char *n
 EXPORT_FUNCTION const char *CALLING_CONVENTION GetCreateLexerError() {
   return errorMessage.c_str();
 }
-}
-
-} // namespace
+} // namespace ScintilluaNS
diff --git a/Scintillua.h b/Scintillua.h
index d349704..8bdbcbd 100644
--- a/Scintillua.h
+++ b/Scintillua.h
@@ -9,19 +9,16 @@
 #ifdef __cplusplus
 #include "ILexer.h"
 #define ILEXER5 Scintilla::ILexer5
-extern "C" {
 #else
 #define ILEXER5 void
 #endif
 
+namespace ScintilluaNS {
 const char *GetLibraryPropertyNames();
 void SetLibraryProperty(const char *key, const char *value);
 const char *GetNameSpace();
 ILEXER5 *CreateLexer(const char *name);
 const char *GetCreateLexerError();
-
-#ifdef __cplusplus
-}
-#endif
+} // namespace ScintilluaNS
 
 #endif
