diff --git a/gzlib.c b/gzlib.c
index dddaf26..68e67fb 100644
--- a/gzlib.c
+++ b/gzlib.c
@@ -2,6 +2,8 @@
  * Copyright (C) 2004-2019 Mark Adler
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
+#include <fcntl.h>
+#include <unistd.h>
 
 #include "gzguts.h"
 
diff --git a/gzread.c b/gzread.c
index 884c9bf..c10fdb9 100644
--- a/gzread.c
+++ b/gzread.c
@@ -2,6 +2,8 @@
  * Copyright (C) 2004-2017 Mark Adler
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
+#include <fcntl.h>
+#include <unistd.h>
 
 #include "gzguts.h"
 
diff --git a/gzwrite.c b/gzwrite.c
index a8ffc8f..05a8512 100644
--- a/gzwrite.c
+++ b/gzwrite.c
@@ -2,6 +2,8 @@
  * Copyright (C) 2004-2019 Mark Adler
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
+#include <fcntl.h>
+#include <unistd.h>
 
 #include "gzguts.h"
 
