diff --git a/gzlib.c b/gzlib.c
index 4105e6a..d94a5c8 100644
--- a/gzlib.c
+++ b/gzlib.c
@@ -2,6 +2,8 @@
  * Copyright (C) 2004-2017 Mark Adler
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
+#include <fcntl.h> // for open
+#include <unistd.h> // for close
 
 #include "gzguts.h"
 
diff --git a/gzread.c b/gzread.c
index 956b91e..9db8a93 100644
--- a/gzread.c
+++ b/gzread.c
@@ -2,6 +2,8 @@
  * Copyright (C) 2004, 2005, 2010, 2011, 2012, 2013, 2016 Mark Adler
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
+#include <fcntl.h> // for open
+#include <unistd.h> // for close
 
 #include "gzguts.h"
 
diff --git a/gzwrite.c b/gzwrite.c
index c7b5651..d569ec9 100644
--- a/gzwrite.c
+++ b/gzwrite.c
@@ -2,6 +2,8 @@
  * Copyright (C) 2004-2017 Mark Adler
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
+#include <fcntl.h> // for open
+#include <unistd.h> // for close
 
 #include "gzguts.h"
 
